examples/v2/cloud_functions/python/cloud_function.py

```diff
@@ -11,54 +11,107 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Creates a cloud function and calls it."""
 
+"""CloudFunction instance-type DM template.
+
+Provisions (or, creates) a pubsub topic.
+"""
+
+import base64
 import json
+import zipfile
+import hashlib
+from StringIO import StringIO
 
 
-def GenerateConfig(context):
+def GenerateConfig(ctx):
   """Generate YAML resource configuration."""
-  deployment_name = context.env['deployment']
-  function_call = deployment_name + '-function-call'
-  function_name = deployment_name + '-function'
-  topic_name = deployment_name + '-topic'
+  inMemoryOutputFile = StringIO()
+  function_name = ctx.env['deployment'] + 'cf'
+  function_call = ctx.env['deployment'] + '-function-call-' + str(ctx.env['current_time'])
+  source_archive_url = ctx.properties['sourceArchiveUrl']
+  zip_file = zipfile.ZipFile(inMemoryOutputFile, mode = 'w', compression = zipfile.ZIP_DEFLATED)
+  for imp in ctx.imports:
+    if imp.startswith('function/'):
+      zip_file.writestr(imp[len('function/'):], ctx.imports[imp])
+  zip_file.close()
+  content = base64.b64encode(inMemoryOutputFile.getvalue())
+  m = hashlib.md5()
+  m.update(content)
 
-  resources = [{
-      'name': topic_name,
-      'type': 'gcp-types/pubsub-v1:projects.topics',
-      'properties': {
-          'topic': topic_name,
-      }
-  }, {
-      'type': 'gcp-types/cloudfunctions-v1beta2:projects.locations.functions',
-      'name': function_name,
-      'properties': {
-          'location': context.properties['region'],
-          'function': function_name,
-          'sourceArchiveUrl': context.properties['sourceArchiveUrl'],
-          'entryPoint': context.properties['entryPoint'],
-          'eventTrigger': {
-              'resource': '$(ref.' + topic_name + '.name)',
-              'eventType': 'providers/cloud.pubsub/eventTypes/topic.publish'
-          }
-      }
-  }, {
-      'name':
-          function_call,
-      'action': ('gcp-types/cloudfunctions-v1beta2:'
-                 'cloudfunctions.projects.locations.functions.call'),
-      'properties': {
-          'name': '$(ref.' + function_name + '.name)',
-          'data': json.dumps({
-              'hola': 'mundo'
-          })
-      }
+  # Simulate a change on the function code by using a different import
```

Comment:
Is this necessary for any regular old cloud function? If so, we should probably make that clearer here. As it is, this seems like a fun line for testing or something.

Reply:
Woups comment was from another change that removed